### PR TITLE
Fix application sync status displaying

### PR DESCRIPTION
**Pull request by cakecatz**
**What this PR does / why we need it**:

Create a component to display application sync status and use it on application list/detail.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/app/web/src/components/app-sync-status.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+}));
+
+interface Props {
+  syncState?: ApplicationSyncState;
+  deploying: boolean;
+  className?: string;
+  size?: "medium" | "large";
+}
+
+const NOT_AVAILABLE_TEXT = "N/A";
```
Comment (khanhtc1202) [inline pkg/app/web/src/components/app-sync-status.tsx:23]: nit, similar to [pkg/app/web/src/components/application-list-item.tsx#L52](https://github.com/pipe-cd/pipe/blob/master/pkg/app/web/src/components/application-list-item.tsx#L52) should we move it to `constants/text.ts` :eyes:

File: pkg/app/web/src/components/application-detail.tsx [hunk 2]
```diff
@@ -63,10 +61,12 @@ const useStyles = makeStyles((theme) => ({
     color: theme.palette.text.secondary,
     marginLeft: theme.spacing(1),
   },
-  syncStatusText: {
-    marginLeft: theme.spacing(0.5),
+  AppSyncState: {
```
Comment (khanhtc1202) [inline pkg/app/web/src/components/application-detail.tsx:64]: nit, do we have conventions for CSS classes' naming, looks like others start with normal character (not capital) 👀

Comment (cakecatz) [reply pkg/app/web/src/components/application-detail.tsx]: Nice catch!

File: pkg/app/web/src/components/app-sync-status.tsx (hunk 1 above)
Comment (cakecatz) [reply pkg/app/web/src/components/app-sync-status.tsx:23]: You're right, thanks😄

**Pull request merged**